### ISSUE TITLE
Typescript - Fetched data is type is not being inferred correctly based on injected data types

### ISSUE THREAD

**Issue by mikeschneiderdotme** [supabase/postgrest-js#407]
# Bug report

## Describe the bug

When querying the database context in Typescript, the data returned is always of type `any` instead inferring the correct type from the injected database types. Previously, the correct types would be returned in the data object.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a database context using type injection as documented [here](https://supabase.com/docs/reference/javascript/typescript-support#injecting-type-definitions)
2. Using the context, query a table with a know injected type
`
const { data, error } = supabaseClient.from('authors').select()
`
3. Observe returned `data` object type

## Expected behavior

The expected behavior, and what previously happened, is that the `data` object type would be correctly inferred as an `Author[]` object based on injected data types. Instead the `data` object is of type `any[]

## Screenshots

![image](https://user-images.githubusercontent.com/45540592/222479643-439ad482-b982-4ce6-ac41-dfe939c37b6d.png)

## System information

- OS: MacOS 13.0.1
- Version of supabase-js: 2.8.0
- Version of Node.js: 16.16.0

**Comment by Frame1910** [comment]
I've got the exact same issue:

<img width="722" alt="Screenshot 2023-03-28 at 10 02 13 am" src="https://user-images.githubusercontent.com/22231370/228107758-4e142518-e60d-4fa0-bcee-92d697371dcf.png">

I've noticed the various PostgrestBuilder objects aren't noticing the typings, however the supabase client object is:

<img width="1009" alt="Screenshot 2023-03-28 at 10 03 32 am" src="https://user-images.githubusercontent.com/22231370/228107937-c9ba6e9f-5921-45dc-97de-3c4a276b58f5.png">

Any insight would be greatly appreciated.

**Comment by galaxyblur** [comment]
Same here. Doesn't work with the Supabase docs' suggestions either.

```
const supabase = createClient<Database>(supabaseUrl, supabaseKey);

export function modelStoreFactory<T extends keyof Database['public']['Tables']>(tableName: string) {
  const tbl = supabase.from(tableName);

  type TRow = Database['public']['Tables'][T]['Row'];
  type TInsert = Database['public']['Tables'][T]['Insert'];

  async function insertItem(item: TInsert): Promise<TRow> {
    const result: PostgrestSingleResponse<TRow> = await tbl.insert(item).select();
  }
}
```

Result:
```
 1. Type 'PostgrestSingleResponse<any[]>' is not assignable to type 'PostgrestSingleResponse<TRow>'.
      Type 'PostgrestResponseSuccess<any[]>' is not assignable to type 'PostgrestSingleResponse<TRow>'.
        Type 'PostgrestResponseSuccess<any[]>' is not assignable to type 'PostgrestResponseSuccess<TRow>'.
          Type 'any[]' is not assignable to type 'TRow'.
```

**Comment by avallete** [comment]
The fix has been released in `supabase-js` v2.46.0. I'm closing this issue, but feel free to reopen if you encounter any further errors.
